### PR TITLE
refactor: use plugin web3 context replace to target chain context

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-2334

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by nuanyang233** [comment]
It is worth noting that when you use web3 context, please make sure that the boundary in the component under the provider is kept in the correct state, for example `ChainBoundary`, sometimes you need to add `PluginWeb3ActualContextProvider` outside the boundary

File: packages/web3-hooks/base/src/useContext.tsx [hunk 1]
```diff
@@ -8,29 +8,33 @@ import { useActualProviderType } from './useProviderType.js'
...
 
-const PluginWeb3Context = createContext<Web3Context<NetworkPluginID>>(EMPTY_OBJECT)
+// Avoid use this context directly
+export const PluginWeb3Context = createContext<Web3Context<NetworkPluginID>>(EMPTY_OBJECT)
 
 const PluginsWeb3Context = createContext<Record<NetworkPluginID, Web3Helper.Web3State<NetworkPluginID>>>(null!)
 
 export function PluginIDContextProvider({ value, children }: React.ProviderProps<NetworkPluginID>) {
+    console.log('------------------useContext PluginIDContextProvider')
+    console.log(value)
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useContext.tsx]:
```suggestion
```

File: packages/web3-hooks/base/src/useContext.tsx [hunk 2]
```diff
@@ -8,29 +8,33 @@ import { useActualProviderType } from './useProviderType.js'
...
 export function PluginWeb3ContextProvider<T extends NetworkPluginID>({
-    pluginID,
     value,
     children,
-}: {
-    pluginID: T
-} & React.ProviderProps<Web3Context<T>>) {
-    return <PluginWeb3Context.Provider value={value} children={children} />
+}: React.ProviderProps<Omit<Web3Context<T>, 'setChainId'>>) {
+    const [chainId, setChainId] = useState<Web3Helper.Definition[T]['ChainId'] | undefined>(value.chainId)
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useContext.tsx:44]: We as well need to set `account`, `providerType`, and `networkType`.

Comment (Lanttcat) [reply packages/web3-hooks/base/src/useContext.tsx:44]: I added the account / providerType setter.  And I think should not care the networkType in Web3ContextProvider.

File: packages/mask/src/components/shared/NetworkTab.tsx [hunk 3]
```diff
@@ -35,7 +34,7 @@ export function NetworkTab<T extends NetworkPluginID = NetworkPluginID.PLUGIN_EV
             <MaskTabList
                 variant="flexible"
                 onChange={(_, v) => {
-                    setChainId(Number.parseInt(v, 10))
+                    setChainId?.(Number.parseInt(v, 10))
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/NetworkTab.tsx]: 👀

File: packages/web3-hooks/base/src/useContext.tsx [hunk 4]
```diff
@@ -8,14 +8,21 @@ import { useActualProviderType } from './useProviderType.js'
 
 interface Web3Context<T extends NetworkPluginID> {
     account?: string
+    networkPluginId?: T
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useContext.tsx]:
```suggestion
    pluginID?: T
```

File: packages/web3-hooks/base/src/useContext.tsx [hunk 5]
```diff
@@ -8,14 +8,28 @@ import { useActualProviderType } from './useProviderType.js'
...
-const PluginWeb3Context = createContext<Web3Context<NetworkPluginID>>(EMPTY_OBJECT)
+const web3ContextInitialState = {
+    setAccount: null!,
+    setChainId: null!,
+    setNetworkType: null!,
+    setProviderType: null!,
+}
+
+// Avoid use this context directly
+export const PluginWeb3Context = createContext<Web3Context<NetworkPluginID>>(web3ContextInitialState)
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useContext.tsx:32]:
```suggestion
export const PluginWeb3Context = createContext<Web3Context<NetworkPluginID>>(null!)
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**